installer/scripts/service_control

```diff
@@ -100,18 +200,26 @@ wait_until_omsagent_stops()
     # Required parameter: Number of seconds to wait for agent to stop
     if [ -z "$1" -o "$1" -le 0 ]; then
         echo "Function \"wait_until_omsagent_stops\" called with invalid parameter"
-        exit 1
+        WS_STATUS=1
+        exit $WS_STATUS
     fi
 
     COUNTER=$(( $1 * 2 )) # Since we sleep 0.5 seconds, compute number of seconds
     while [ $COUNTER -gt 0 ]; do
-        is_omsagent_running && return $?
-        COUNTER=$(( $COUNTER - 1 ))
-        sleep 0.5
+        if is_omsagent_running; then
+            COUNTER=$(( $COUNTER - 1 ))
+            sleep 0.5
+        else
+            remove_PIDfile_unless_omsagent_running
+            WS_STATUS=$? # Should only be non-zero of the removal of a PIDFILE failed for some reason.
```

Comment:
The `remove_PIDfile_unless_omsagent_running` method is somewhat confusing; you say here that it should return 0 if the PIDFILE was correctly deleted, but you return 1 [here](https://github.com/Microsoft/OMS-Agent-for-Linux/pull/573/files#diff-c1d234f8111f79a335f4309684025d80R186) when the rm command returns 0. Please comment on and validate the expected behavior.